internal/character/danhengimbibitorlunae/skill.go

```diff
@@ -0,0 +1,82 @@
+package danhengimbibitorlunae
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	SkillEffect  = "danhengimbibitorlunae-skill-effect"
+	EnhanceLevel = "danhengimbibitorlunae-enhancelevel"
+)
+
+type skillState struct {
+	CritDMGBoost float64
+}
+
+// enhance attack,has 3 type,use 1/2/3 skill point
+
+func (c *char) initSkill() {
+	modifier.Register(EnhanceLevel, modifier.Config{
+		Stacking:          modifier.ReplaceBySource,
+		MaxCount:          3,
+		StatusType:        model.StatusType_UNKNOWN_STATUS,
+		CountAddWhenStack: 1,
+	})
+	// if E4, the effect has 1 more turn and can be refreshed
+	if c.info.Eidolon < 4 {
```

Comment:
Remove the conditional and just do one Register, when AddModifier is called for SkillEffect, check if E4 then Duration = 2 and finally add  Duration inside the info.Modifier. In this case the AddModifier for this modifier is in the AddSkill function.

This is how the Register would look like here.
```
modifier.Register(SkillEffect, modifier.Config{
			StatusType: model.StatusType_STATUS_BUFF,
			Stacking: modifier.ReplaceBySource,
			TickMoment: modifier.ModifierPhase2End
			MaxCount: 4,
			Duration: 1,
			Listeners: modifier.Listeners{
				OnAdd: SkillOnAdd,
			},
			CountAddWhenStack: 1,
		})
```